src/framework/components/layout-group/layout-calculator.js

```diff
@@ -0,0 +1,596 @@
+                var actualSize = Math.min(targetSize, maxSize);
+
+                sizesThisLine[index][axis.size] = actualSize;
+
+                // Work out how much of the total undershoot value we've just used,
+                // and decrement the remaining value by this much.
+                var actualIncrease = Math.max(targetSize - actualSize, 0);
+                var appliedIncrease = targetIncrease - actualIncrease;
+
+                remainingUndershoot -= appliedIncrease;
+            }
+        }
+
+        // This loop is very similar to the one in stretchSizesToFitContainer() above,
+        // but with some awkward inversions and use of min as opposed to max etc that
+        // mean a more generalized version would probably be harder to read/debug than
+        // just having a small amount of duplication.
+        function shrinkSizesToFitContainer(sizesThisLine, idealRequiredSpace, axis) {
+            var descendingMinSizeOrder = getTraversalOrder(sizesThisLine, axis.minSize, true);
+            var fittingProportions = getNormalizedValues(sizesThisLine, axis.fittingProportion);
+            var inverseFittingProportions = invertNormalizedValues(fittingProportions);
+            var inverseFittingProportionSums = createSumArray(inverseFittingProportions, descendingMinSizeOrder);
+
+            var remainingOvershoot = idealRequiredSpace - availableSpace[axis.axis];
+
+            for (var i = 0; i < sizesThisLine.length; ++i) {
+                var index = descendingMinSizeOrder[i];
+
+                // Similar to the stretch calculation above, we calculate the ideal
+                // size reduction value for this element based on its fitting proportion.
+                //
+                // However, note that we're using the inverse of the fitting value, as
+                // using the regular value would mean that an element with a fitting
+                // value of, say, 0.4, ends up rendering very small when shrinking is
+                // being applied. Using the inverse means that the balance of sizes
+                // between elements is similar for both the Stretch and Shrink modes.
+                var targetReduction = calculateAdjustment(index, remainingOvershoot, inverseFittingProportions, inverseFittingProportionSums);
+                var targetSize = sizesThisLine[index][axis.size] - targetReduction;
+
+                var minSize = sizesThisLine[index][axis.minSize];
+                var actualSize = Math.max(targetSize, minSize);
+
+                sizesThisLine[index][axis.size] = actualSize;
+
+                var actualReduction = Math.max(actualSize - targetSize, 0);
+                var appliedReduction = targetReduction - actualReduction;
+
+                remainingOvershoot -= appliedReduction;
+            }
+        }
+
+        function calculateAdjustment(index, remainingAdjustment, fittingProportions, fittingProportionSums) {
+            const proportion = fittingProportions[index];
```

Comment:
`const` keywords are not currently allowed with our linting rules

Reply:
Ah, sorry – force of habit :) will fix.